workflow-ui/compose/src/main/java/com/squareup/workflow1/ui/compose/RenderAsState.kt

```diff
@@ -206,18 +206,23 @@ private class WorkflowRuntimeState<PropsT, OutputT : Any, RenderingT>(
       onOutput = onOutput
     )
 
-    renderings
-      .onEach {
-        renderingState.value = it.rendering
-        snapshotState.value = it.snapshot
-      }
+    workflowScope.launch(
+      start = UNDISPATCHED,
+      context = Dispatchers.Unconfined
+    ) {
```

Comment:
moved from shuttling these renderings and snapshot into compose state in an `onEach` started by a `launchIn` on `Dispatchers.Unconfined` to launching for the collection with an `CoroutineStart.UNDISPATCHED` coroutine and this fixes the timing issue we saw come up in the Compose sample test while scrolling the LazyColumn.

cc @rjrjr and @RBusarow .

I discussed this with @zach-klippenstein . He thinks there may still be some Compose bug here but this is making the behavior more reliable for us for now and while it makes this PR messy I'm inclined to merge it.

Reply:
It seems fine to me 👍

Reply:
This could be a big win. The problem your describing sounds very similar to one that has been plaguing a lot of our Compose tests.

Reply:
Oh, but we don't use `renderAsState` :(